third_party/nvidia/hopper/lib/Transforms/WarpSpecialization/CodePartitionUtility.cpp

```diff
@@ -578,8 +626,16 @@ void getBufferIdxAndPhase(OpBuilderWithAsyncTaskIds &builder, Operation *op,
   }
   // Increment accumCnt if there are multiple channels in the reuseGroup in this
   // region.
-  Value idxVal = builder.createWithAsyncTaskIds<arith::ConstantIntOp>(
-      op->getLoc(), theIdx, 64);
+  // Create idxVal with the same type as accumCnt to ensure type compatibility
+  Value idxVal;
+  if (accumCnt.getType().isIndex()) {
```

Comment:
I don't have a great answer for the implementation, but seems like a lot of special casing on indexing. I wonder if we can generate a generic (deriveType) function that can handle indexing rules based on the desired output type/bit width.